components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1421,6 +1392,15 @@ void RewardsServiceImpl::OnAttestPromotion(
         brave_promotion);
   }
 
+  if (promotion->type == ledger::PromotionType::ADS) {
+    auto* ads_service = brave_ads::AdsServiceFactory::GetForProfile(profile_);
+    if (ads_service) {
```

Comment:
`ads_service->UpdateAdRewards(/*should_reconcile*/ true);` need to be in this if and not `return`. You can't return here if ads_don't exist as there is rewards logic below

Reply:
Good spot :-) fixed